### ISSUE TITLE
Spotify problem

### ISSUE THREAD

**Issue by day4me** [qtile/qtile#1792]
The problem is that you can't open the dropdown menu in Spotify/Settings by clicking it with mouse or touchpad, but you can open it with keyboard. It works in different wm's(like awesome).

[https://streamable.com/hr2fg4](url)

You can see in the video, I tried to open dropdown menu by clicking it with my mouse, but it only worked when i opened it with keybord.

qtile version - 0.15.1
OS: arch linux
kernel: 5.7.5
spotify ver - 1:1.1.10.546-4 (aur)
x.org x server ver - 1.20.8

**Comment by tych0** [comment]
On Thu, Jun 25, 2020 at 11:37:38AM -0700, day4me wrote:
> The problem is that you can't open the dropdown menu in Spotify/Settings by clicking it with mouse or touchpad, but you can open it with keyboard. It works in different wm's(like awesome).
> 
> [https://streamable.com/hr2fg4](url)
> 
> You can see in the video, I tried to open dropdown menu by clicking it with my mouse, but it only worked when i opened it with keybord.
> 
> qtile version - 0.15.1
> OS: arch linux
> kernel: 5.7.5
> spotify ver - 1:1.1.10.546-4 (aur)
> x.org x server ver - 1.20.8

Yeah, I see the same behavior. It's odd too, since the dropdown isn't
a second window, so it doesn't seem like qtile should be interfering,
but somehow it is.

**Comment by ramnes** [comment]
Ah, funny, I've always thought this was a problem from the Linux version of the application itself, not Qtile. The workaround is to put your cursor on the menu item you want to click and then press Enter.

**Comment by tych0** [comment]
On Fri, Jun 26, 2020 at 12:00:49AM -0700, Guillaume Gelin wrote:
> Ah, funny, I've always thought this was a problem from the Linux version of the application itself, not Qtile. The workaround is to put your cursor on the menu item you want to click and then press Enter.

Yeah, I sort of thought the same thing since it wasn't a separate
window. I never bothered to try it in other WMs though, so if it works
there, there's some qtile bug...

**Comment by day4me** [comment]
> On Fri, Jun 26, 2020 at 12:00:49AM -0700, Guillaume Gelin wrote: Ah, funny, I've always thought this was a problem from the Linux version of the application itself, not Qtile. The workaround is to put your cursor on the menu item you want to click and then press Enter.
> Yeah, I sort of thought the same thing since it wasn't a separate window. I never bothered to try it in other WMs though, so if it works there, there's some qtile bug...

yeah, i thought the same, because it doesn't work in openbox and herbstluftwm too(for me), but then i tried awesome, bspwm and dwm and it works there.

**Comment by ramnes** [comment]
Stop answering via email @tych0, I deleted my comment and now it's back, eh! In fact I thought the issue was about the top right arrow menu, which AFAIK doesn't work with any distribution, until I watched the video, and realized this was about the menus inside the settings. And these one do look like separate windows, @tych0.

**Comment by ramnes** [comment]
@day4me It looks like I can open the menus by not releasing the click immediately, but after half a second or so. Do you get this behavior also?

**Comment by tych0** [comment]
On Fri, Jun 26, 2020 at 09:06:44AM -0700, Guillaume Gelin wrote:
> And these one do look like separate windows, @tych0.

They do? They're not for me... can you show an xprop of the dialog?

**Comment by day4me** [comment]
> @day4me It looks like I can open the menus by not releasing the click immediately, but after half a second or so. Do you get this behavior also?

No, menu just get triggered by clicking and not releasing the click immediately does nothing for me.
And also about the top right arrow menu, yeah it doesn't work too, but it actually works in awesome or bspwm. And i think this two issues are related(maybe).

**Comment by day4me** [comment]
> On Fri, Jun 26, 2020 at 09:06:44AM -0700, Guillaume Gelin wrote: And these one do look like separate windows, @tych0.
> They do? They're not for me... can you show an xprop of the dialog?
[https://imgur.com/FxdFZ5o](https://imgur.com/FxdFZ5o)
first xprop is regular spotify window
second one is the dialog(menu)

**Comment by tych0** [comment]
On Fri, Jun 26, 2020 at 09:42:10AM -0700, day4me wrote:
> > On Fri, Jun 26, 2020 at 09:06:44AM -0700, Guillaume Gelin wrote: And these one do look like separate windows, @tych0.
> > They do? They're not for me... can you show an xprop of the dialog?
> [https://imgur.com/FxdFZ5o](https://imgur.com/FxdFZ5o)
> first xprop is regular spotify window
> second one is the dialog(menu)

Can you paste those as text here? :)

**Comment by day4me** [comment]
> On Fri, Jun 26, 2020 at 09:42:10AM -0700, day4me wrote: > On Fri, Jun 26, 2020 at 09:06:44AM -0700, Guillaume Gelin wrote: And these one do look like separate windows, @tych0. > They do? They're not for me... can you show an xprop of the dialog? [[https://imgur.com/FxdFZ5o](https://imgur.com/FxdFZ5o)](https://imgur.com/FxdFZ5o%5D(https://imgur.com/FxdFZ5o)) first xprop is regular spotify window second one is the dialog(menu)
> Can you paste those as text here? :)

`[day4me@arch-d4m ~]$ xprop 
_NET_WM_ICON(CARDINAL) = 
WM_CLASS(STRING) = "spotify", "Spotify"
WM_NAME(STRING) = "Spotify Premium"
_NET_WM_NAME(UTF8_STRING) = "Spotify Premium"
XdndProxy(WINDOW): window id # 0x800002
_NET_WM_DESKTOP(CARDINAL) = 0
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 0, 0
		window gravity: Static
_NET_WM_PID(CARDINAL) = 334
WM_LOCALE_NAME(STRING) = "uk_UA.UTF-8"
WM_CLIENT_MACHINE(STRING) = "arch-d4m"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, _NET_WM_PING
[day4me@arch-d4m ~]$ xprop 
_NET_WM_USER_TIME(CARDINAL) = 0
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 349, 309
XdndAware(ATOM) = BITMAP
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x0, 0x0, 0x0
_NET_WM_BYPASS_COMPOSITOR(CARDINAL) = 2
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_MENU
_NET_WM_PID(CARDINAL) = 334
WM_LOCALE_NAME(STRING) = "uk_UA.UTF-8"
WM_CLIENT_MACHINE(STRING) = "arch-d4m"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, _NET_WM_PING`

**Comment by tych0** [comment]
On Fri, Jun 26, 2020 at 10:29:21AM -0700, day4me wrote:
> [day4me@arch-d4m ~]$ xprop 
> _NET_WM_USER_TIME(CARDINAL) = 0
> WM_NORMAL_HINTS(WM_SIZE_HINTS):
> 		program specified location: 349, 309

This suggests that something like
cbb21efe2ed94355c98b4dd228aebdaa5f1a1e9c may be the culprit. Can you
try with that reverted and see?

**Comment by day4me** [comment]
> On Fri, Jun 26, 2020 at 10:29:21AM -0700, day4me wrote: ***@***.*** ~]$ xprop _NET_WM_USER_TIME(CARDINAL) = 0 WM_NORMAL_HINTS(WM_SIZE_HINTS): program specified location: 349, 309
> This suggests that something like [cbb21ef](https://github.com/qtile/qtile/commit/cbb21efe2ed94355c98b4dd228aebdaa5f1a1e9c) may be the culprit. Can you try with that reverted and see?
> […](#)
> XdndAware(ATOM) = BITMAP _MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x0, 0x0, 0x0 _NET_WM_BYPASS_COMPOSITOR(CARDINAL) = 2 _NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_MENU _NET_WM_PID(CARDINAL) = 334 WM_LOCALE_NAME(STRING) = "uk_UA.UTF-8" WM_CLIENT_MACHINE(STRING) = "arch-d4m" WM_PROTOCOLS(ATOM): protocols WM_DELETE_WINDOW, _NET_WM_PING` -- You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub: [#1792 (comment)](https://github.com/qtile/qtile/issues/1792#issuecomment-650303056)

i'm not using the master git version, i'm using the last release 0.15.1.
but i installed git version and it did nothing, same behaviour.

**Comment by Thesald** [comment]
I confirm the same issue on my setup (Arch Linux 5.10.7-arch1-1 kernel, latest stable Qtile 0.16.1-2, Nvidia gpu with latest proprietary drivers (there are always a possibility that Nvidia will mess up something unrelated)) But I remember getting to the settings when Spotify prompted a message at top saying it was in auto play mode and can be changed in settings. When I clicked to that, it did open settings and I was able to change settings (download directory, streaming quality etc.) However, that was only time I was able to get in. I think that Qtile have nothing to do with that since it can be opened by different methods. It might be a bug on Spotify side.

**Comment by m-col** [comment]
Does https://github.com/qtile/qtile/pull/2506 fix it?
